apps/nextjs-x-chain/src/app/components/CCTPWithdraw.tsx

```diff
@@ -0,0 +1,381 @@
+import { useCallback, useEffect, useState } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardTitle,
+  CardContent,
+  CardHeader,
+  CardDescription,
+} from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { ChainConfig, CrossChainCore } from "@aptos-labs/cross-chain-core";
+import {
+  Account,
+  Ed25519PrivateKey,
+  Network,
+  PrivateKey,
+  PrivateKeyVariants,
+} from "@aptos-labs/ts-sdk";
+import {
+  Chain,
+  WormholeTransferResponse,
+  WormholeQuoteResponse,
+} from "@aptos-labs/cross-chain-core";
+import { AdapterWallet } from "@aptos-labs/wallet-adapter-core";
+import { Loader2, MoveDown } from "lucide-react";
+import USDC from "@/app/icons/USDC";
+import { chainToIcon } from "@/app/icons";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import {
+  isEIP1193DerivedWallet,
+  OriginWalletDetails,
+} from "@/utils/derivedWallet";
+import { isSolanaDerivedWallet } from "@/utils/derivedWallet";
+import { useUSDCBalance } from "@/contexts/USDCBalanceContext";
+
+const dappNetwork: Network.MAINNET | Network.TESTNET = Network.TESTNET;
+
+const crossChainCore = new CrossChainCore({
+  dappConfig: { aptosNetwork: dappNetwork },
+});
+const provider = crossChainCore.getProvider("Wormhole");
+
+let mainSigner: Account;
+let sponsorAccount: Account;
+
+// Set the main signer account
+const mainSignerPrivateKey =
+  process.env.NEXT_PUBLIC_SWAP_CCTP_MAIN_SIGNER_PRIVATE_KEY ||
```

Comment:
I don't have access to the secrets but is this set for the deployed version?

Reply:
yes, we store the sponsor account in a vault